File: backend/src/models/intel.py
"""
统一运营情报数据模型
"""

from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class IntelSourceType(str, Enum):
    """统一来源类型"""

    XIAOHONGSHU = "xiaohongshu"
    UCAS = "ucas"
    UNIVERSITY_SITE = "university_site"
    EXAM_BOARD = "exam_board"
    VISA_POLICY = "visa_policy"
    WECHAT_MEDIA = "wechat_media"


class IntelSourceSyncStatus(str, Enum):
    """来源同步状态。"""

    SUCCESS = "success"
    BLOCKED = "blocked"
    ERROR = "error"


class IntelItem(BaseModel):
    """统一情报项"""

    item_id: str
    source_type: IntelSourceType
    source_name: str
    title: str
    summary_short: str
    summary_long: str
    impact_targets: list[str] = Field(default_factory=list)
    published_at: datetime
    collected_at: datetime
    original_url: str
    priority_hint: str | None = None
    school_name: str | None = None
    source_group: str | None = None
    external_id: str | None = None
    content_html: str | None = None


class IntelOverviewSection(BaseModel):
    """总览页来源分区"""

    source_key: str
    source_label: str
    total_items: int
    preview_items: list[IntelItem]


class IntelHelperRail(BaseModel):
    """右侧轻辅助栏"""

    highlight_count: int
    top_counts: dict[str, int]


class IntelSourceSyncReport(BaseModel):
    """单个来源的一次同步结果。"""

    source_id: str
    source_type: IntelSourceType
    source_name: str
    school_name: str | None = None
    status: IntelSourceSyncStatus
    item_count: int = 0
    error_message: str | None = None
    notes: str | None = None
    synced_at: datetime

File: frontend/src/types/intel.ts
export type IntelSourceKey =
  | "overview"
  | "ucas"
  | "university_site"
  | "exam_board"
  | "visa_policy"
  | "wechat_media";

export type IntelSourceFeedKey = Exclude<IntelSourceKey, "overview">;

export type IntelAdapterType =
  | "feed"
  | "listing"
  | "rss"
  | "json_feed"
  | "html_listing";

export interface ListingSelectors {
  item: string;
  title: string;
  url: string;
  summary?: string | null;
  date?: string | null;
}

export interface IntelSourceConfig {
  source_id: string;
  source_type: IntelSourceFeedKey;
  school_name?: string | null;
  adapter_type: IntelAdapterType;
  source_group: string;
  source_name: string;
  feed_url?: string | null;
  listing_url?: string | null;
  selectors?: ListingSelectors | null;
  enabled?: boolean;
}

export interface ProbeIntelSourceRequest {
  url: string;
  source_type: IntelSourceFeedKey;
  source_name?: string | null;
  source_group: string;
}

export type ProbeIntelSourceStatus = "success" | "blocked" | "unsupported";

export interface ProbeIntelSourceResponse {
  status: ProbeIntelSourceStatus;
  message: string;
  sample_count: number;
  recommended_source: IntelSourceConfig;
}

export type CreateIntelSourceRequest = IntelSourceConfig;

export interface CreateIntelSourceResponse {
  source: IntelSourceConfig;
  item_count: number;
  sync_report: IntelSourceSyncReport;
}

export interface IntelItem {
  item_id: string;
  source_type: IntelSourceFeedKey;
  source_name: string;
  title: string;
  summary_short: string;
  summary_long: string;
  impact_targets: string[];
  published_at: string;
  collected_at: string;
  original_url: string;
  school_name?: string;
  source_group?: string;
}

export type IntelSourceSyncStatus = "success" | "blocked" | "error";

export interface IntelSourceSyncReport {
  source_id: string;
  source_type: IntelSourceFeedKey;
  source_name: string;
  school_name?: string;
  status: IntelSourceSyncStatus;
  item_count: number;
  error_message?: string | null;
  notes?: string | null;
  synced_at: string;
}

export interface IntelOverviewSection {
  source_key: IntelSourceFeedKey;
  source_label: string;
  total_items: number;
  preview_items: IntelItem[];
}

export interface IntelHelperRail {
  highlight_count: number;
  top_counts: Record<string, number>;
}

export interface IntelOverviewResponse {
  sections: IntelOverviewSection[];
  helper_rail: IntelHelperRail;
}

export interface IntelSourceResponse {
  source_key: IntelSourceFeedKey;
  items: IntelItem[];
  helper_rail: IntelHelperRail;
  sync_reports: IntelSourceSyncReport[];
}

export interface IntelSourceSyncStatusResponse {
  source_key: IntelSourceFeedKey;
  reports: IntelSourceSyncReport[];
}

export interface SourceNavItem {
  key: IntelSourceKey;
  label: string;
}
